src/engine/resolvers.rs

```diff
@@ -80,6 +81,65 @@ where
         }
     }
 
+    /// Returns the resolver input deserialized into a structure of type T that
+    /// implements the serde `Deserialize` trait.
```

Comment:
Is the idea to give people a convenience function/method that lets them work with data in a native Rust struct, rather than in serialized Warpgrapher `Value` types?

Reply:
Correct. For several reasons. One, deserializing to a struct is a good way to guarantee that you have all the fields that you are expecting in an input, and it provides a perfect way to fail fast on a request and return some type of InvalidInput error. Second, after deserializing into a struct, you can directly access fields with `mystruct.some_field` instead of `input.get("some_field").unwrap().as_string().unwrap()` except with error handling instead of unwrap.

Reply:
That makes sense.  Nice.